src/bootstrap/setup.rs

```diff
@@ -107,6 +108,19 @@ pub fn setup(src_path: &Path, profile: Profile) {
     let include_path = profile.include_path(src_path);
     println!("`x.py` will now use the configuration at {}", include_path.display());
 
+    let build = TargetSelection::from_user(&env!("BUILD_TRIPLE"));
+    let stage_path = ["build", build.rustc_target_arg(), "stage1"].join("/");
+    let toolchain_link =
+        Command::new("rustup").args(&["toolchain", "link", "stage1", &stage_path[..]]).output();
+    if toolchain_link.is_err() {
+        println!("WARNING: `rustup`failed to link stage 1 build to `stage1` toolchain");
+        println!(
+            "To manually link stage 1 build to `stage1` tool chain, run:\n
+            `rustup toolchain link stage1 build/{}/stage1`",
+            build.rustc_target_arg()
```

Comment:
So... I could have missed something about the previous code, but at this point in the program, what proof do we have that the user **has** Rustup on their system that justifies emitting this message all "WARNING!" style, telling them about a binary that possibly isn't installed on their system? It is entirely possible that the user is bootstrapping Rust on their system for the first time, Rustup is not available for this target, and we should be careful to neither alarm nor distract them. Something gentler and softer-voiced seems appropriate.

Reply:
If rustup is installed:
```
Linking stage1 toolchain
```

If not:
```
rustup is not installed
Unable to link stage1 toolchain
```

Reply:
I think it would be fine to only warn if the profile isn't `"user"` (which is what distros should be using); for everyone else they would find rustup really useful for rust-lang/rust even if they wouldn't use it normally.

Reply:
It's mostly about framing: gently observing that they do so is fine, as long as it doesn't appear on every invocation after, if they decide they don't want to use it.

Reply:
@workingjubilee yup - this is only running once when you first run `setup`, it's never repeated.